src/server/serveractiveobject.cpp

```diff
@@ -29,6 +30,13 @@ ServerActiveObject::ServerActiveObject(ServerEnvironment *env, v3f pos):
 {
 }
 
+void ServerActiveObject::setBasePosition(v3f pos) {
+	bool changed = m_base_position != pos;
+	m_base_position = pos;
+	if (changed && getEnv()) // HACK *when* is getEnv() null?
```

Comment:
Potentially cleaner alternatives:

* Couple active objects to their manager by giving them a direct reference; only allow object creation through the manager.
* Get rid of `setBasePosition`, force position updates to go through the manager that way. The manager can befriend the `m_base_position` member variable.